Get top warriors

Fesmofet opened the pull request.

Fesmofet committed:
simple update with no validation

Fesmofet committed:
Merge branch 'development' into update-user-info

Fesmofet committed:
add validation

Fesmofet committed:
add documentation update-user-info and add example of responses to all routes

Fesmofet committed:
userModel test

Fesmofet committed:
test failed at token verify

Fesmofet committed:
test

Fesmofet committed:
test

Fesmofet committed:
invalid token

Fesmofet committed:
user controller tests

Fesmofet committed:
logout and swagger

Fesmofet committed:
change const

Fesmofet committed:
change battle schema add avatar and alias

Fesmofet committed:
test userModel/destroySession

Fesmofet committed:
controller tests

Fesmofet committed:
crypto schema

Fesmofet committed:
start model

Fesmofet committed:
checkout

Fesmofet committed:
battle controller check error

Fesmofet committed:
logout error handle user controller error handle

Fesmofet committed:
Merge branch 'change-battle-schema' into logout

Fesmofet committed:
Merge branch 'logout' into get-crypto

Fesmofet committed:
model test

Fesmofet committed:
add documentation

Fesmofet committed:
rout

Fesmofet committed:
message error

Fesmofet committed:
Merge branch 'logout' into get-top-warriors

Fesmofet committed:
handle errors + tests

Fesmofet committed:
add documentation

Fesmofet committed:
correct test

Fesmofet committed:
refactor

crypto-wars-wd commented on controllers/authenticationController.js:
```diff
@@ -35,8 +36,26 @@ const hasSocialAccount = async (req, res) => {
 };
 
+const logout = async (req, res) => {
+  let user, message;
+
+  const { params, validationError } = validators.validate(req.body, validators.authentication.logoutShcema);
+  if (validationError) return render.error(res, validationError);
+
+  const { payload, error } = await sessions.getAuthData({ req });
+  if (error) return render.unauthorized(res, error);
+
+  ({ user, message } = await userModel.findUserById(params.id));
+  if (message) return render.error(res, message);
+
+  const session = sessions.findSession({ sessions: user && user.auth && user.auth.sessions, sid: payload.sid });
+  await userModel.destroySession({ userId: user._id, session });
+  return render.success(res);
```
Очень много логики в контроллере, вынеси в операции и здесь вызывай один метод из операции. Зачем ты сделал user & message let? они не переопределяются, смотри красное подчеркивание от Eslint

Fesmofet replied:
done

crypto-wars-wd commented on controllers/authenticationController.js:
```diff
@@ -5,6 +5,7 @@ const { userModel } = require('models');
 const render = require('concerns/render');
 const { strategies } = require('utilities/operations');
 const { setAuthHeaders } = require('utilities/authentication/sessions');
+const { sessions } = require('utilities/authentication');
```
что это такое? ты 2 раза импортишь одно и то же, по сути

Fesmofet replied:
done

crypto-wars-wd commented on controllers/battleController.js:
```diff
@@ -1,16 +1,32 @@
-  const result = await battleModel.createNewBattle(params);
 
-  return render.success(res, result);
+  const { battle, message } = await battleModel.createNewBattle(params);
```
Почему ты везде используешь название обьекта ошибки message?

Fesmofet replied:
done

Fesmofet committed:
virtual don't create

Fesmofet committed:
change battle schema, swagger, battle controller, battle model add constants, add battle operations

Fesmofet committed:
refactor battle controller  and battleSchema

Fesmofet committed:
small refactor auth controller

Fesmofet committed:
end has more

Fesmofet committed:
add region routes

Fesmofet committed:
refactor creating

Fesmofet committed:
before handaling error change

Fesmofet committed:
success handle error

Fesmofet committed:
hasMore added

Fesmofet committed:
add task create crypto in mongodb

Fesmofet committed:
test fix

Fesmofet committed:
change documentation add limit and skip to top-warrioirs

Fesmofet committed:
replace message to errorSignUp

Fesmofet committed:
uninstall pug axios form-data

Fesmofet committed:
change schema

Fesmofet committed:
add constants to schema

Fesmofet committed:
empty line

Fesmofet committed:
rename  methods

Fesmofet committed:
fix handle error

Fesmofet committed:
uninstall rewire

Fesmofet committed:
test except lodash crypto model

Fesmofet committed:
change task to async

Fesmofet committed:
console.log

Fesmofet committed:
refactor path to operations

Fesmofet committed:
change populate path add test with random

Fesmofet committed:
add swager

Fesmofet committed:
error in models

Fesmofet committed:
refactor

Fesmofet committed:
change handle error

Fesmofet committed:
CRYPTO_KEY in config

Fesmofet committed:
add 503 error
